### ISSUE TITLE
how do we pass edge option from serenity.property file ?

### ISSUE THREAD

**Issue by varmavg** [serenity-bdd/serenity-cucumber#246]
We are getting pop up message “we couldn’t load the extensions from:” when trying to open Edge (chromium). 

We handled this in chrome like  chrome.switches = -disable-extensions 

Just wondering if there is there anything equivalent to chrome.switches  in Edge chromium to provide options in sereity.properties ?

![popup](https://user-images.githubusercontent.com/49928330/90830261-5b0fd400-e30f-11ea-8710-fbd3b0487e16.png)

**Comment by Chelinadevi** [comment]
Hi,

I am working on serenity framework for UI automation, and I need to run my test on Edge browser as well, there are few which I am facing when I run on edge browser. I am using msedgedriver.exe(Chromium)
The error i am getting is could not instantiate Webdriver.

Could you pls help with the edge options that I need to use in serenity properties file?

Thanks in Advance

**Comment by kiranAdiga** [comment]
@varmavg were you able to run yours tests successfully on microsoft edge chromium browser successfully.?

**Comment by varmavg** [comment]
yes we were able to run successful tests in Edge chromium. use edge chromium 32 bit driver (x86 version) https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/

**Comment by kiranAdiga** [comment]
@varmavg Thanks for the quick response.
Can you let me know what is that you passed in serenity.conf file or anything that you would have done to run your tests on edge chromium browser using serenity .

**Comment by varmavg** [comment]
(1) POM.xml here is my configurations 

	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<maven.compiler.source>1.8</maven.compiler.source>
		<maven.compiler.target>1.8</maven.compiler.target>
		<serenity.version>2.0.40</serenity.version>
		<serenity.maven.version>2.0.40</serenity.maven.version>
		<serenity.cucumber.version>1.9.31</serenity.cucumber.version>
		<tags></tags>
		<parallel.tests>4</parallel.tests>
		<webdriver.base.url></webdriver.base.url>
		<webdriver.driver>edge</webdriver.driver>
		
		<parallel.tests>3</parallel.tests>
	</properties>


(2) Serenity.properties 

# chrome
webdriver.chrome.driver=C:\\Automation\\Drivers\\chromedriver.exe
chrome.switches=--start-maximized --ignore-urlfetcher-cert-requests --ignore-certificate-errors --disable-machine-cert-request -disable-extensions
# edge
webdriver.edge.driver=C:\\Automation\\Drivers\\msedgedriver.exe


(3) installed 32bit Edge drive (compatible with you edge version) in above location. 

(4) run configuration 
clean verify

**Comment by kiranAdiga** [comment]
@varmavg Thanks for the response.
Looks like in the latest versions of serenity ,running scenarios on chromium edge browser is not possible.

**Comment by moebius23** [comment]
@kiranAdiga Have you tried to remove the path of the Edge webdriver of the _serenity.conf_. And put the following code in the object inheriting from PageObject and that all of your custom PageObject inherit from : 

`
    static{
        System.setProperty("webdriver.edge.driver", "your/path/to/the/webdriver/msedgedriver.exe");
    }
`
It worked for me.

**Comment by mrjcohen** [comment]
Does anyone know if this has been fixed as to be able to pass edgedriver within the serenity.config file like we do with other web drivers?

**Comment by juanquiceno2000** [comment]
@moebius23 Hello could you help me please with the solution that you shared before??
I Should set this 
static{ System.setProperty("webdriver.edge.driver", "your/path/to/the/webdriver/msedgedriver.exe");
 } 
Where exactly??? I'm new in SerenityBDD and I need to run a couple of tests in Edge.
Thanks.

**Comment by juanquiceno2000** [comment]
> Does anyone know if this has been fixed as to be able to pass edgedriver within the serenity.config file like we do with other web drivers?

No, is not working through Serenity.confing file.

**Comment by moebius23** [comment]
> @moebius23 Hello could you please help me with the solution that you shared before?? I Should set this static{ System.setProperty("webdriver.edge.driver", "your/path/to/the/webdriver/msedgedriver.exe"); } Where exactly??? I'm new in SerenityBDD and I need to run a couple of tests on Edge. Thanks.
So in my _serenity.conf_ file, I have set "webdriver.driver = edge".

In my project, every part of the front have its PageObject inheriting from _net.serenitybdd.core.pages.PageObject_ (example: _NavigationPage_, _HeaderPage_, _ProductsPage_, and so on). But they don't inherit directly. I have an object than inherit from _PageObject_, and every specialised Page inherit from. Let's call it _MyProjectPage_. 

Then the inheritance path should look like this:
_PageObject_->_MyProjectPage_->ProductsPage_

I have put the **static block** that we are talking about inside _MyProductPage_ class, at the beginning.

I hope it will work for you !

**Comment by juanquiceno2000** [comment]
> > @moebius23 Hello could you please help me with the solution that you shared before?? I Should set this static{ System.setProperty("webdriver.edge.driver", "your/path/to/the/webdriver/msedgedriver.exe"); } Where exactly??? I'm new in SerenityBDD and I need to run a couple of tests on Edge. Thanks.
> > So in my _serenity.conf_ file, I have set "webdriver.driver = edge".
> 
> In my project, every part of the front have its PageObject inheriting from _net.serenitybdd.core.pages.PageObject_ (example: _NavigationPage_, _HeaderPage_, _ProductsPage_, and so on). But they don't inherit directly. I have an object than inherit from _PageObject_, and every specialised Page inherit from. Let's call it _MyProjectPage_.
> 
> Then the inheritance path should look like this: _PageObject_->_MyProjectPage_->_ProductsPage_
> 
> I have put the **static block** that we are talking about inside _MyProductPage_ class, at the beginning.
> 
> I hope it will work for you !

@moebius23 Thank you Sir!!!!! It worked perfect!!!

**Comment by juan2three** [comment]
Is there any update on this?

**Comment by wakaleo** [comment]
> Is there any update on this?

It's an open source project: look in the code (EdgeDriverCapabilities and [WhenConfiguringEdgeOptions](https://github.com/serenity-bdd/serenity-core/blob/1f5316739594593b2b1c630c3734291752259c35/serenity-core/src/test/groovy/net/thucydides/core/webdriver/edge/WhenConfiguringEdgeOptions.groovy), get an example working with your code, write up some documentation, and submit a PR to the documentation project.
